dips/dip-5.md

```diff
@@ -0,0 +1,80 @@
+
+## Specification
+The change will involve all components of the DeSo stack, including Core, Backend, Identity, and Frontend. In addition, app developers should adjust how they communicate with the DeSo Identity Service to include the messaging party in message decryption. Here are Github pull requests for reference:
+- [DeSo core](https://github.com/deso-protocol/core/pull/170)
+- [DeSo backend](https://github.com/deso-protocol/backend/pull/253)
+- [DeSo identity](https://github.com/deso-protocol/identity/pull/107)
+- [DeSo node frontend](https://github.com/deso-protocol/frontend/pull/543)
+
+### Technical changes
+
+#### Core
+A conditional statement will be added in `_connectBasicTransfer()` to look for transaction's `ExtraData` and check for three fields: `MessagingPublicKey`, `MessagingKeyName`, and `MessagingKeySignature`. These respectively will contain the public key and its name along with an owner-signed hash of `(MessagingPublicKey + MessagingKeyName)`. The signature is provided because we want derived key holders to submit messaging keys; therefore, the transaction signature alone wouldn't be sufficient to prove authorization of the messaging key. Similarly, disconnecting a messaging key will happen in `_disconnectBasicTransfer()` and will delete the messaging entry from the `UtxoView` and DB. Core will only accept one key for a label, an attempt to submit a different public key for the same key name will silently error.
+
+We also change how `PrivateMessage` transactions are connected. To send a V3 message, a `PrivateMessage` transaction should have `Version: 3` field in metadata, as well as `ExtraData` containing a message party information comprising `SenderMessagingPublicKey`, `SenderMessagingKeyName`, `RecipientMessagingPublicKey`, and `RecipientMessagingKeyName`. These fields will describe the messaging keys for the sender and recipient used when encrypting the private message. The fields will be added by core in `_connectPrivateMessage()` which will use them to construct a message and message party entries. Removing the entries from `UtxoView` or DB will happen in `_disconnectPrivateMessage()`.
+
+#### Backend
+
+We modify both `/api/v0/get-messages-stateless` and `/api/v0/send-message-stateless` endpoints to include the messaging party information. We also add an endpoint to fetch user messaging keys.
+
+#### Identity
```

Comment:
I'd like to see some information about how the keys are derived deterministically.